Adapters/Vungle/ISVungleAdapter/ISVungleAdapter.m

```diff
@@ -44,35 +45,11 @@ typedef NS_ENUM(NSInteger, InitState) {
 static InitState _initState = INIT_STATE_NONE;
 static ConcurrentMutableSet<ISNetworkInitCallbackProtocol> *initCallbackDelegates = nil;
 
-// Banner state possible values
-typedef NS_ENUM(NSUInteger, BANNER_STATE) {
-    UNKNOWN,
-    REQUESTING,
-    REQUESTING_RELOAD,
-    SHOWING
-};
-
-@interface ISVungleAdapter() < VungleDelegate, ISNetworkInitCallbackProtocol, InitiatorDelegate> {
-}
+@interface ISVungleAdapter() <ISNetworkInitCallbackProtocol>
 
-// Rewarded video
-@property (nonatomic, strong) ConcurrentMutableDictionary *rewardedVideoPlacementIdToSmashDelegate;
-@property (nonatomic, strong) ConcurrentMutableDictionary *rewardedVideoPlacementIdToServerData;
-@property (nonatomic, strong) ConcurrentMutableDictionary *rewardedVideoServerDataToSmashDelegate;
-@property (nonatomic, strong) ConcurrentMutableSet        *rewardedVideoPlacementIdsForInitCallbacks;
-
-// Interstitial
-@property (nonatomic, strong) ConcurrentMutableDictionary *interstitialPlacementIdToSmashDelegate;
-@property (nonatomic, strong) ConcurrentMutableDictionary *interstitialPlacementIdToServerData;
-@property (nonatomic, strong) ConcurrentMutableDictionary *interstitialServerDataToSmashDelegate;
-
-// Banner
-@property (nonatomic, strong) ConcurrentMutableDictionary *bannerPlacementIdToSmashDelegate;
-@property (nonatomic, strong) ConcurrentMutableDictionary *bannerPlacementIdToSize;
-@property (nonatomic, strong) ConcurrentMutableDictionary *bannerPlacementIdToViewController;
-@property (nonatomic, strong) ConcurrentMutableDictionary *bannerPlacementIdToBannerState;
-@property (nonatomic, strong) ConcurrentMutableDictionary *bannerPlacementIdToServerData;
-@property (nonatomic, strong) ConcurrentMutableDictionary *bannerServerDataToSmashDelegate;
+@property (nonatomic, strong) ISVungleRewardedVideoAdapterRouter *rewardedVideoAdapterRouter;
+@property (nonatomic, strong) ISVungleInterstitialAdapterRouter *interstitialAdapterRouter;
```

Comment:
@YueVungle I'm unfamiliar with the ironSource dashboard. Does this mean that in ironSource, there can only be 1 interstitial ad unit at any time for 1 app id?
There's no way to have two or more interstitial placements for 1 app id?

Reply:
Hi @stanley-vungle, we can create several ad units for each ad type in IronSource Dashboard. Because for Interstitial or Rewarded Ad, it only can load and play one Ad at a time.  After closing the old one, we can load a new one.
For Banner ad, SDK will not show more than one banner at a time. 

So my idea is that for each ad type, there should be only one instance in memory at a time. When there is a new ad request from mediation, for example Interstitial Ad, `interstitialAdapterRouter` of `ISVungleAdapter` class will point to a new instance corresponding to this new ad request, the old one is already closed :
```
- (void)initInterstitialWithUserId:(NSString *)userId
                     adapterConfig:(ISAdapterConfig *)adapterConfig
                          delegate:(id<ISInterstitialAdapterDelegate>)delegate {
    NSString *appId = adapterConfig.settings[kAppID];
    NSString *placementId = adapterConfig.settings[kPlacementID];
    ...

    LogInternal_Internal(@"appId = %@, placementId = %@", appId, placementId);
    self.interstitialAdapterRouter = [[ISVungleInterstitialAdapterRouter alloc] initWithPlacementID:placementId parentAdapter:self delegate:delegate];
    ...
    }
```
Thanks a lot. :)